protocol/rpcprovider/rewardserver/rewards_store.go

```diff
@@ -0,0 +1,169 @@
+package rewardserver
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/utils/sigs"
+	pairingtypes "github.com/lavanet/lava/x/pairing/types"
+	"golang.org/x/exp/slices"
+)
+
+const keySeparator = "."
+
+type RewardStore struct {
+	db DB
+}
+
+type RewardEntity struct {
+	Epoch        uint64
+	ConsumerAddr string
+	ConsumerKey  string
+	SessionId    uint64
+	Proof        *pairingtypes.RelaySession
+}
+
+func (rs *RewardStore) Save(ctx context.Context, consumerAddr string, consumerKey string, proof *pairingtypes.RelaySession) (bool, error) {
+	key := assembleKey(uint64(proof.Epoch), consumerAddr, proof.SessionId, consumerKey)
+
+	re := &RewardEntity{
+		Epoch:        uint64(proof.Epoch),
+		ConsumerAddr: consumerAddr,
+		ConsumerKey:  consumerKey,
+		SessionId:    proof.SessionId,
+		Proof:        proof,
+	}
+
+	buf, err := json.Marshal(re)
+	if err != nil {
+		return false, utils.LavaFormatError("failed to encode proof: %s", err)
+	}
+
+	rs.db.Save(ctx, key, buf)
+
+	return true, nil
+}
+
+func (rs *RewardStore) FindOne(
+	ctx context.Context,
+	epoch uint64,
+	consumerAddr string,
+	consumerKey string,
+	sessionId uint64,
+) (*pairingtypes.RelaySession, error) {
+	key := assembleKey(epoch, consumerAddr, sessionId, consumerKey)
+
+	rawReward, err := rs.db.FindOne(ctx, key)
+	if err != nil {
+		return nil, utils.LavaFormatDebug("reward not found")
+	}
+
+	var re RewardEntity
+	err = json.Unmarshal(rawReward, &re)
+	if err != nil {
+		return nil, utils.LavaFormatError("failed to decode proof: %s", err)
+	}
+
+	return re.Proof, nil
+}
+
+func (rs *RewardStore) FindAll(ctx context.Context) (map[uint64]*EpochRewards, error) {
+	rawRewards, err := rs.db.FindAll(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[uint64]*EpochRewards)
+	for key, rewards := range rawRewards {
+		epoch, consumerAddr, sessionId, consumerKey := keyParts(key)
```

Comment:
we dont need to extract all this information from the key, as we already unmarshal the information from the rewards object and get the rewardsEntity, which already contains the information we need. (epoch , session id, consumerkey , consumer address..